pallet/account-migration/src/lib.rs

```diff
@@ -173,8 +185,158 @@ pub mod pallet {
+		pub fn migrate_multisig(
+			origin: OriginFor<T>,
+			who: AccountId32,
+			others: Vec<AccountId32>,
+			threshold: u16,
+			to: AccountId20,
+			_signature: Signature,
+		) -> DispatchResult {
+			ensure_none(origin)?;
+
+			let (members, multisig) = multisig_of(who.clone(), others, threshold);
+
+			if threshold < 2 {
+				Self::migrate_inner(multisig, to)?;
+			} else {
+				let mut members = members.into_iter().map(|m| (m, false)).collect::<Vec<_>>();
+
+				// Set the status to `true`.
+				//
+				// Because the `_signature` was already been verified in `pre_dispatch`.
+				members
+					.iter_mut()
+					.find(|(a, _)| a == &who)
+					.expect("[pallet::account-migration] `members` will never be empty; qed")
+					.1 = true;
+
+				<Multisigs<T>>::insert(multisig, Multisig { migrate_to: to, members, threshold });
+			}
+
+			Ok(())
+		}
+
+		/// To complete the pending multisig migration.
+		///
+		/// The `_signature` should be provided by `submitter`.
+		#[pallet::call_index(2)]
+		#[pallet::weight(0)]
+		pub fn complete_multisig_migration(
+			origin: OriginFor<T>,
+			multisig: AccountId32,
+			submitter: AccountId32,
+			_signature: Signature,
+		) -> DispatchResult {
+			ensure_none(origin)?;
+
+			let mut multisig_info = <Multisigs<T>>::take(&multisig)
+				.expect("[pallet::account-migration] already checked in `pre_dispatch`; qed");
+
+			// Set the status to `true`.
+			//
+			// Because the `_signature` was already been verified in `pre_dispatch`.
+			multisig_info
+				.members
+				.iter_mut()
+				.find(|(a, _)| a == &submitter)
+				.expect("[pallet::account-migration] already checked in `pre_dispatch`; qed")
+				.1 = true;
+
+			if multisig_info.members.iter().fold(0, |acc, (_, ok)| if *ok { acc + 1 } else { acc })
+				>= multisig_info.threshold
+			{
+				Self::migrate_inner(multisig, multisig_info.migrate_to)?;
+			} else {
+				<Multisigs<T>>::insert(multisig, multisig_info);
+			}
+
+			Ok(())
+		}
+	}
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+
+		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			match call {
+				Call::migrate { from, to, signature } => {
+					Self::pre_check_existing(from, to)?;
+
+					Self::pre_check_signature(from, to, signature)
+				},
+				Call::migrate_multisig { who, others, threshold, to, signature } => {
+					let (_, multisig) = multisig_of(who.to_owned(), others.to_owned(), *threshold);
+
+					Self::pre_check_existing(&multisig, to)?;
+
+					Self::pre_check_signature(who, to, signature)
+				},
+				Call::complete_multisig_migration { multisig, submitter, signature } => {
+					let Some(multisig_info) = <Multisigs<T>>::get(multisig) else {
+						return InvalidTransaction::Custom(E_ACCOUNT_NOT_FOUND).into();
+					};
+
+					if !multisig_info.members.iter().any(|(a, _)| a == submitter) {
+						return InvalidTransaction::Custom(E_NOT_MULTISIG_MEMBER).into();
+					}
```

Comment:
I think we should reject useless requests from members whose status is already `true`.